src/Microsoft.Graph.Core/Requests/Content/BatchResponseContent.cs

```diff
@@ -0,0 +1,146 @@
+            if (jBatchResponseObject == null)
+                return null;
+
+            JObject jResponseItem = null;
+
+            if (jBatchResponseObject.TryGetValue("responses", out JToken jResponses))
+            {
+                jResponseItem = jResponses.FirstOrDefault((jtoken) => jtoken.Value<string>("id").Equals(requestId)) as JObject;
+            }
+
+            return GetResponseMessageFromJObject(jResponseItem);
+        }
+
+        /// <summary>
+        /// Gets the @NextLink of a batch response.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<string> GetNextLinkAsync()
+        {
+            jBatchResponseObject = jBatchResponseObject ?? await GetBatchResponseContentAsync();
+            if (jBatchResponseObject == null)
+                return null;
+
+            return jBatchResponseObject.GetValue("@nextLink")?.ToString();
+        }
+
+        /// <summary>
+        /// Gets a <see cref="HttpResponseMessage"/> from <see cref="JObject"/> representing a batch response item.
+        /// </summary>
+        /// <param name="jResponseItem">A single batch response item of type <see cref="JObject"/>.</param>
+        /// <returns>A single batch response as a <see cref="HttpResponseMessage"/>.</returns>
+        private HttpResponseMessage GetResponseMessageFromJObject(JObject jResponseItem)
+        {
+            if (jResponseItem == null)
+                return null;
+
+            HttpResponseMessage responseMessage = new HttpResponseMessage();
+
+            if (jResponseItem.TryGetValue("status", out JToken status))
+            {
+                responseMessage.StatusCode = (HttpStatusCode)int.Parse(status.ToString());
+            }
+
+            if (jResponseItem.TryGetValue("body", out JToken body))
+            {
+                responseMessage.Content = new StringContent(body.ToString(), Encoding.UTF8, "application/json");
+            }
+
+            if (jResponseItem.TryGetValue("headers", out JToken headers))
+            {
+                foreach (KeyValuePair<string, string> headerKeyValue in headers.ToObject<Dictionary<string, string>>())
+                {
+                    responseMessage.Headers.TryAddWithoutValidation(headerKeyValue.Key, headerKeyValue.Value);
+                }
+            }
+            return responseMessage;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="HttpContent"/> of a batch response as <see cref="JObject"/>.
+        /// </summary>
+        /// <returns>A batch response content as <see cref="JObject"/>.</returns>
+        private async Task<JObject> GetBatchResponseContentAsync()
+        {
+            if (this.batchResponseMessage.Content == null)
+                return null;
+
+            try
+            {
+                string content = await this.batchResponseMessage.Content.ReadAsStringAsync();
+                return JsonConvert.DeserializeObject<JObject>(content);
```

Comment:
When you do these two lines of code, you are reading bytes off the wire, loading them into a buffer, converting it into a string, passing the string in JsonConvert, which will then load the string into a JsonReader and attempt to use it's generic deserialization method to load the string into a JObject.   
Or you could ReadAsStreamAsync, put the stream in a StreamReader, pass it to a JsonTextReader and call JObject.Load and the bytes will come right off the wire, straight into a JObject.  No magic needed :-)